src/peft/tuners/adalora/model.py

```diff
@@ -244,7 +252,17 @@ def forward(self, *args, **kwargs):
             num_param = 0
             for n, p in self.model.named_parameters():
                 if ("lora_A" in n or "lora_B" in n) and self.trainable_adapter_name in n:
-                    para_cov = p @ p.T if "lora_A" in n else p.T @ p
+                    if is_deepspeed_zero3_enabled():
```

Comment:
the input of [gather_params_ctx](https://github.com/huggingface/peft/blob/26726bf1ddee6ca75ed4e1bfd292094526707a78/src/peft/utils/integrations.py#L23) is torch.nn.Module. not nn.Parameters.

Reply:
Hello @sywangyi, maybe we can make agruments of `gather_params_ctx` as kwargs, something like below and use it:
```
def gather_params_ctx(module: torch.nn.Module = None, parameter: torch.nn.Parameter = None, modifier_rank: int = 0)
    """Call DeepSpeed GatheredParameters context manager if DeepSpeed is enabled, otherwise do nothing."""
    if packaging.version.parse(transformers.__version__) >= packaging.version.parse("4.33.0"):
        from transformers.integrations import is_deepspeed_zero3_enabled
    else:
        from transformers.deepspeed import is_deepspeed_zero3_enabled

    if not is_deepspeed_zero3_enabled():
        yield
        return

    import deepspeed
    if module is None and parameter is None:
        raise ValueError("At least one of the arguments `module` or `parameter` should be valid, both are None.")

    params_to_gather = list(module.parameters()) if module is not None else []
    params_to_gather = params_to_gather.append(parameter) if parameter is not None else params_to_gather
    
    with deepspeed.zero.GatheredParameters(params_to_gather, modifier_rank=modifier_rank):
        yield
    return
```